src/Source.cpp

```diff
@@ -335,6 +335,87 @@ void SourceUniformCylinder::setDescription() {
 	description = ss.str();
 }
 
+// ---------------------------------------------------------------------------
+SourceSNRDistribution::SourceSNRDistribution() :
+    R_earth(8.5*kpc), beta(3.53), Zg(0.3*kpc) {
+	set_frMax(8.5*kpc, 3.53);
+	set_fzMax(0.3*kpc);
+}
+
+SourceSNRDistribution::SourceSNRDistribution(double R_earth, double beta, double Zg) :
+    R_earth(R_earth), beta(beta), Zg(Zg) {
+	set_frMax(R_earth, beta);
+	set_fzMax(Zg);
+}
+
+void SourceSNRDistribution::prepareParticle(ParticleState& particle) const {
+  	Random &random = Random::instance();
+	double RPos;
+	while (true){
+		RPos = random.rand()*20.*kpc;
```

Comment:
Is there any reason why R_max is hard-coded to 20*kpc? At least, for clarity, I would put it in the header file and here would put R_max. The same comment goes for ZPos a few lines below.